File: sdk/python/src/lmsservice_db/authentication.py
import json

class AuthenticationService:
    def __init__(self, client):
        self.client = client

    def authenticate_user(self, username, password):
        endpoint = 'users/login-password'
        payload = {
            'UserName': username,
            'Password': password
        }
        response = self.client.make_request('POST', endpoint, data=payload)

        if 'Data' in response and 'AccessToken' in response['Data']:
            self.client.set_access_token(response['Data']['AccessToken'])
        else:
            raise ValueError(f"Token not found in response: {response}")
        
        formatted_response = json.dumps(response, indent=4)
        print("Authenticated : ", formatted_response)
        
        return response

File: sdk/node/src/api/authentication.ts

import { APIClient } from './client';

interface AuthResponse {
    Data: {
        AccessToken: string;
        RefreshToken?: string;
        ExpiresIn?: number;
        TokenType?: string;
    };
    Message?: string;
    Success?: boolean;
}

export default class AuthenticationService {
    private APIClient: APIClient;

    constructor(APIClient: APIClient) {
        this.APIClient = APIClient;
    }

    async authenticateUser(username: string, password: string): Promise<AuthResponse> {
        const endpoint = 'users/login-password';
        const payload = { UserName: username, Password: password };
        
        const response = await this.APIClient.makeRequest<AuthResponse>('POST', endpoint, payload);

        if (response.Data?.AccessToken) {
            this.APIClient.setAccessToken(response.Data.AccessToken);
        } else {
            throw new Error('Authentication failed: Token not found in response.');
        }
        
        console.log('Authentication response:', response);
        return response;
    }
}